python/demo/demo_navier_stokes.py

```diff
@@ -275,7 +275,7 @@ def jump(phi, n):
     pressure_dof = [pressure_dofs[0]]
 else:
     pressure_dof = []
-bc_p = fem.dirichletbc(0.0, np.array(pressure_dof, dtype=np.int32), Q)
+bc_p = fem.dirichletbc(PETSc.ScalarType(0.0), np.array(pressure_dof, dtype=np.int32), Q)
```

Comment:
This is ugly, especially since the demo doesn't support complex types (due to a bunch of UFL shortcomings). Why is it required for mypy?

Reply:
`float` isn't included in the typing list for `dirichletbc`. I'll revert the change to this PR and make a new PR against dolfinx with float added